codeql/windows-drivers/queries/Windows/wdk/wdm/SAL.qll

```diff
@@ -0,0 +1,221 @@
+/**
+ * Provides classes for identifying and reasoning about Microsoft source code
+ * annotation language (SAL) macros.
+ * 
+ * This version of this file has been updated to include driverspecs.h as a valid SAL header.
+ * It will be removed if and when that change is upstreamed to the main CodeQL repo.
```

Comment:
Valid was a poor choice of words.  The default SAL library in the CodeQL repo does not refer to driverspecs.h as it is a header only meaningful to Windows drivers.  In the past we have gotten some pushback when asking if we should bring Windows driver-specific code into the main CodeQL repo, but I can open an issue to get more feedback from them on this specific case.  

For TODOs, do you know if we generally publicize AzDO work item numbers in public code? That strikes me as a little odd, but I could be overthinking it.

Reply:
If you are concerned about the potential security risk of publishing a task number then I'd advise you to ask the appropriate teams, otherwise I think it is wise to be able to track the items that aren't finished yet and provide a small indication of such to the public.

Reply:
TODO: Add a TODO: to the comment block to be more clear.